src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -106,7 +106,7 @@ private BoundExpression CompareWithConstant(BoundExpression input, BoundExpressi
                 );
         }
 
-        BoundExpression MakeDeclarationPattern(SyntaxNode syntax, BoundExpression input, LocalSymbol target, bool requiresNullTest)
+        BoundExpression MakeDeclarationPattern(SyntaxNode syntax, BoundExpression input, BoundExpression target, bool requiresNullTest)
```

Comment:
I'd rather not encode knowledge of the EE into pattern lowering. That kind of cross-component coupling makes the code more fragile than it needs to be.

---
In reply to: [77736440](https://github.com/dotnet/roslyn/pull/13535#discussion_r77736440) [](ancestors = 77736440,77735261,77712197)